### DISCUSSION TITLE
本身是公网IP的情况下不能玩吗？

### DISCUSSION THREAD

**Discussion by xsj684135** [MikeWang000000/Natter#57] [Q&A]
本身有公网的，通过ikuai拨号的
TCP hole punching seems to fail. Maybe you are behind a firewall. However, you may check this address from another network: ('113.67.177.29', 6789)

**Comment by xsj684135** [comment]
[INFO] - Getting STUN server IP...
[INFO] - Checking NAT Type for UDP...
[INFO] - NAT Type for UDP: [ Full cone (NAT 1) ]
[INFO] - Checking NAT Type for TCP...
[INFO] - NAT Type for TCP: [ Cone NAT ]
[INFO] - Start punching...

**Comment by MikeWang000000** [comment]
可以加上 `-v` 查看更详细的日志吗？您的公网 IP 可能是“假公网” (ref: https://v2ex.com/t/875867)
理论公网 IP 应该能看到 `NAT Type for UDP: [ Open Internet ]`，而不是 NAT 1。

**Comment by xsj684135** [comment]
[INFO] - Getting STUN server IP...
[DEBUG] - Resolving hostname [fwa.lifesizecloud.com]...
[DEBUG] - Resolving hostname [stun.isp.net.au]...
[DEBUG] - Resolving hostname [stun.freeswitch.org]...
[DEBUG] - Resolving hostname [stun.voip.blackberry.com]...
[DEBUG] - Resolving hostname [stun.nextcloud.com]...
[DEBUG] - Resolving hostname [stun.stunprotocol.org]...
[DEBUG] - Resolving hostname [stun.sipnet.com]...
[DEBUG] - Resolving hostname [stun.radiojar.com]...
[DEBUG] - Resolving hostname [stun.sonetel.com]...
[DEBUG] - Resolving hostname [stun.voipgate.com]...
[DEBUG] - Resolving hostname [stun.miwifi.com]...
[DEBUG] - Resolving hostname [stun.qq.com]...
[INFO] - Checking NAT Type for UDP...
[DEBUG] - Trying UDP STUN: 111.206.174.2 (change ip:0/port:0)
[DEBUG] - (UDP) ('111.206.174.2', 3478) says: ('113.67.177.29', 17950)
[DEBUG] - Trying UDP STUN: 111.206.174.3 (change ip:0/port:0)
[DEBUG] - (UDP) ('111.206.174.3', 3478) says: ('113.67.177.29', 17950)
[DEBUG] - Trying UDP STUN: 111.206.174.3 (change ip:1/port:1)
[DEBUG] - (UDP) ('111.206.174.2', 3479) says: ('113.67.177.29', 17950)
[DEBUG] - Trying UDP STUN: 111.206.174.3 (change ip:0/port:1)
[DEBUG] - (UDP) ('111.206.174.3', 3479) says: ('113.67.177.29', 17950)
[INFO] - NAT Type for UDP: [ Full cone (NAT 1) ]
[INFO] - Checking NAT Type for TCP...
[DEBUG] - Trying TCP STUN: 54.195.170.93
[DEBUG] - (TCP) 54.195.170.93 says: ('113.67.177.29', 59704)
[DEBUG] - Trying TCP STUN: 34.245.212.183
[DEBUG] - (TCP) 34.245.212.183 says: ('113.67.177.29', 59704)
[DEBUG] - Trying TCP STUN: 13.229.64.74
[DEBUG] - (TCP) 13.229.64.74 says: ('113.67.177.29', 59704)
[INFO] - NAT Type for TCP: [ Cone NAT ]
[INFO] - Start punching...
[DEBUG] - Trying TCP STUN: 54.195.170.93
[DEBUG] - (TCP) 54.195.170.93 says: ('113.67.177.29', 3456)
[WARNING] - TCP hole punching seems to fail. Maybe you are behind a firewall. However, you may check this address from another network: ('113.67.177.29', 3456)
[INFO] - TCP keep-alive...
[DEBUG] - [Thu Sep 15 18:16:37 2022] Keep-Alive OK!

**Comment by xsj684135** [comment]
需要在ikuai上做一次端口映射吗？或者设置DMZ？

**Comment by MikeWang000000** [comment]
> 需要在ikuai上做一次端口映射吗？或者设置DMZ？

看到了，您是在局域网的主机上打洞的，那么要在ikuai上对打洞的主机设DMZ。

不过既然您已经有公网了，那么DMZ就直接暴露在外了，没有必要使用Natter了吧。

**Comment by fbion** [comment]
> > 需要在ikuai上做一次端口映射吗？或者设置DMZ？
> 
> 看到了，您是在局域网的主机上打洞的，那么要在ikuai上对打洞的主机设DMZ。
> 
> 不过既然您已经有公网了，那么DMZ就直接暴露在外了，没有必要使用Natter了吧。

如果没有路由器权限呢- -#

**Comment by YYQHoro** [comment]
> > > 需要在ikuai上做一次端口映射吗？或者设置DMZ？
> > 
> > 
> > 看到了，您是在局域网的主机上打洞的，那么要在ikuai上对打洞的主机设DMZ。
> > 不过既然您已经有公网了，那么DMZ就直接暴露在外了，没有必要使用Natter了吧。
> 
> 如果没有路由器权限呢- -#

可以使用UPnP设备或软件对路由器配置端口映射，如果路由器支持UPnP。此方法不需要路由器的账号密码。
